enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,180 @@
+        Blocking too many edges is better than blocking too few, because you can recover from the former, but possibly brick clusters on the latter.
+
+        The result of the query may be cached by the update service, so queries which generate unstable result sets should be avoided.
+        Clusters whose submitted `id` is not in the result set may be considered cache misses and, if a cache refresh still fails to include them, may be cached as `blocked` for the usual cache duration.
+
+[The schema version][graph-data-schema-version] would also be bumped to 1.1.0, because this is a backwards-compatible change.
+Consumers who only understand graph-data schema 1.0.0 would ignore the `clusters` property and block the edge for all clusters.
+While blocking the edge for all clusters is more aggressive than the graph-data maintainers intended, it is failing on the side of safety.
+Blocking these edges for all clusters is also not as aggressive as complaining about an unrecognized 2.0.0 version and failing to serve the entire graph.
+
+### Update service support for the enhanced schema
+
+The following recommendations are geared towards the [openshift/cincinnati][cincinnati].
+Maintainers of other update services may or may not be able to apply them to their own implementation.
+
+The graph-builder's graph-data scraper should learn about [the new 1.1.0 schema](#enhanced-graph-data-schema-for-blocking-edges), and record any `clusters` properties in edge metadata.
+
+When a request is received, the submitted [`channel` query parameter][cincinnati-for-openshift-request] limits the set of remaining edges.
+If any of the remaining edges have `clusters.prometheus` entries, the update service will exclude the edge if the [`id` query parameter][cincinnati-for-openshift-request] is a member of the query's resulting ID set.
+To perform the PromQL request, the update service will be extended with configuration for the Prometheus connection, including a URI and authentication credentials.
+
+To reliably and efficiently return cluster-specific responses, the ID set returned by configured queries may be cached.
+There are a few possibilities for caching; although selecting a particular caching implementation is up to the update service's maintainers.
+The policy engine should refresh the cache on cache misses, as discussed in [the schema section](#enhanced-graph-data-schema-for-blocking-edges).
+
+For every graph-builder response, the policy engine may aggregate queries for all edges and warm the cache of new queries.
+This will make the initial client request faster, at the expense of maintaining a cache that may not be needed before it goes stale.
+
+The policy engine should also be prepared for, and alert on, misconfigured PromQL that results in request failures.
+In the event of such a failure, the fallback behavior should be blocking the relevant edge for all clusters until the misconfiguration is fixed.
+
```

Comment:
We would need a way to track which cluster are blocked by this schema? Otherwise we might forget about the clusters and they might never upgrade to a new version.

Reply:
We could expose a metric with blocked cluster IDs, so that we could have an option to check if a particular cluster is blocked (it would be nice to mention which rule caused it too)

Reply:
I think metrics are overkill for this, and have added a section on *Stranding supported clusters* in b78cced -> 2d2b1de with discussion.  Also a new section on *Denial of service via ID flooding*.